common/static/common/js/components/collections/paging_collection.js

```diff
@@ -0,0 +1,218 @@
+                // to the same object references for sortableFields and filterableFields.
+                this.sortableFields = {};
+                this.filterableFields = {};
+            },
+
+            paginator_core: {
+                type: 'GET',
+                dataType: 'json',
+                url: function () { return this.url; }
+            },
+
+            paginator_ui: {
+                firstPage: function () { return this.isZeroIndexed ? 0 : 1; },
+                currentPage: function () { return this.isZeroIndexed ? 0 : 1; },
+                perPage: function () { return this.perPage; }
+            },
+
+            server_api: {
+                'page': function () { return this.currentPage; },
+                'page_size': function () { return this.perPage; },
+                'sort_order': function () { return this.sortField; }
+            },
+
+            parse: function (response) {
+                this.totalCount = response.count;
+                this.currentPage = response.current_page;
+                this.totalPages = response.num_pages;
+                this.start = response.start;
+                return response.results;
+            },
+
+            isZeroIndexed: false,
+            perPage: 10,
+
+            sortField: '',
+            sortDirection: 'descending',
+            sortableFields: {},
+
+            filterField: '',
+            filterableFields: {},
+
+            /**
+             * Returns the current page number as if numbering starts on page one, regardless of the indexing of the
+             * underlying server API.
+             */
+            currentOneIndexPage: function () {
+                return this.currentPage + (this.isZeroIndexed ? 1 : 0);
+            },
+
+            /**
+             * Sets the current page of the collection. Page is assumed to be one indexed, regardless of the indexing
+             * of the underlying server API. If there is an error fetching the page, the Backbone 'error' event is
+             * triggered and the page does not change. A 'page_changed' event is triggered on a successful page change.
+             * @param page one-indexed page to change to
+             */
+            setPage: function (page) {
+                var oldPage = this.currentPage,
+                    self = this;
+                this.goTo(page - (this.isZeroIndexed ? 1 : 0), {
+                    reset: true,
+                    success: function () {
+                        self.trigger('page_changed');
+                    },
+                    error: function () {
+                        self.currentPage = oldPage;
+                    }
+                });
+            },
+
+            /**
+             * Returns true if the collection has a next page, false otherwise.
+             */
+            hasNextPage: function () {
+                return this.currentOneIndexPage() + 1 <= this.totalPages;
```

Comment:
I think this would be clearer as:

    this.currentOneIndexPage() < this.totalPages;